### ISSUE TITLE
metadata validation failing for dandiset 000971

### ISSUE THREAD

**Issue by bendichter** [dandi/dandi-archive#1958]
https://dandiarchive.org/dandiset/000971/draft

We are currently seeing the following errors:

<img width="834" alt="image" src="https://github.com/dandi/dandi-archive/assets/844306/a5021ecb-329d-4e43-aae1-0f0a619f18fa">

Looking at the metadata, I am not sure what the problem is.

1. Is this an error in metadata validation?
2. Is there any way we could improve these error messages to be more informative? In particular, it is not clear which contributor is failing validation here, so it is difficult to identify and fix the error for metadata that has many contributors. Could we add this information? I think I would be able to determine this from the full json-schema traceback. Maybe we could add a "details" button by each validation error which shows this traceback.

cc @pauladkisson

**Comment by waxlamp** [comment]
Hey @bendichter, I agree that we need to do a lot better with these error messages.

In the current case, I also cannot find where the errors are. I'm going to look into what's going on so the specific case can be resolved (while retaining general ideas for making error reporting significantly more useful in the future).

**Comment by waxlamp** [comment]
@bendichter: so it turns out that Joseph M. Bianco's entry in the contributor list has a stray comma that should be removed; this extra comma was causing the name to not comport with the regular expression. (Side note: it continues to amaze me what escapes the human eye in matters like this--I truly couldn't see that extra comma as I scanned over the list of contributors! If only there were a scientific field that studied such matters...)

The second and third errors result from the validator then thinking "hrm, maybe this failed validation because it's not really a Person, but an Organization", and finding errors related to that assumption (i.e., if this is an Organization then indeed the `schemaKey` field should read `Organization`, and it should have an ROR ID). This is a bug in the validation logic, since we essentially need the system to trust that the `schemaKey` field is always correct (and therefore, never generate errors based on other prospective validation passes). @mvandenburgh is already working on a fix for this.

If you fix the contributor entry and the validation errors go away, I'll close this issue in favor of one that addresses the need for better error messages in general. Mike and I collected some intel that will help and I will post it there.
